solidity/random-beacon/contracts/libraries/DKG.sol

```diff
@@ -245,116 +259,92 @@ library DKG {
-    ///      calculated group public key, misbehaved members as bytes and DKG
-    ///      start block. The calculated hash should be prefixed with prefixed with
-    ///      `\x19Ethereum signed message:\n` before signing, so the message to
-    ///      sign is:
-    ///      `\x19Ethereum signed message:\n${keccak256(groupPubKey,misbehaved,startBlock)}`
-    ///      Members indexing in the group starts with 1.
-    /// @param result DKG result which will be verified.
-    function verify(Data storage self, Result calldata result) internal view {
-        // TODO: Verify if submitter is valid staker and signatures come from valid
-        // stakers https://github.com/keep-network/keep-core/pull/2654#discussion_r728226906.
-
-        require(result.submitterMemberIndex > 0, "Invalid submitter index");
-
+    /// @notice Performs basic validation of the submitted DKG result.
+    function validateSubmittedResult(Data storage self, Result calldata result)
+        internal
+        view
+    {
         SortitionPool sortitionPool = self.sortitionPool;
 
+        // Submitter must be an operator in the sortition pool.
+        // Declared submitter's member index in the DKG result needs to match
+        // the address calling this function.
+        require(
+            sortitionPool.isOperatorInPool(msg.sender),
+            "Submitter not in the sortition pool"
+        );
         require(
             sortitionPool.getIDOperator(
                 result.members[result.submitterMemberIndex - 1]
             ) == msg.sender,
             "Unexpected submitter index"
         );
 
-        uint256 T_init = self.startBlock +
-            offchainDkgTime +
-            self.resultSubmissionStartBlockOffset;
-        require(
-            block.number >=
-                (T_init +
-                    (result.submitterMemberIndex - 1) *
-                    self.parameters.resultSubmissionEligibilityDelay),
-            "Submitter not eligible"
-        );
-
+        // Group public key needs to be 128 bytes long.
         require(result.groupPubKey.length == 128, "Malformed group public key");
 
+        // The number of misbehaved members can not exceed the threshold.
+        // Misbehaved member indices needs to be unique, between [1,64],
+        // and sorted in ascending order.
+        uint8[] calldata misbehavedMembersIndices = result
+            .misbehavedMembersIndices;
         require(
-            groupSize - result.misbehavedMembersIndices.length >=
-                activeThreshold,
+            groupSize - misbehavedMembersIndices.length >= activeThreshold,
             "Too many members misbehaving during DKG"
         );
-
-        if (result.misbehavedMembersIndices.length > 1) {
-            for (
-                uint256 i = 1;
-                i < result.misbehavedMembersIndices.length;
-                i++
-            ) {
+        if (misbehavedMembersIndices.length > 1) {
```

Comment:
I think we could rewrite it like this:
```       
if (misbehavedMembersIndices.length > 1) {
    require(
        misbehavedMembersIndices[0] >= 1 &&
            misbehavedMembersIndices[
                misbehavedMembersIndices.length - 1
            ] <=
            groupSize,
        "Corrupted misbehaved members indices"
    );

    for (uint256 i = 0; i < misbehavedMembersIndices.length - 1; i++) {
        require(
            misbehavedMembersIndices[i] <
                misbehavedMembersIndices[i + 1],
            "Corrupted misbehaved members indices"
        );
    }
}
```

If we ensure the elements are sorted, then we just need to check the left-most and right-most values are `>=1 && <= groupSize`.

Reply:
Yep, very good catch!